components/componentsWrapper/ComponentsWrapper.tsx

```diff
@@ -0,0 +1,11 @@
+import { ReactElement } from "react";
+
+export interface IComponentsWrapper{
+    condition: boolean;
+    ReturnComponent: ReactElement;
+    AlternativeComponent: ReactElement; 
+}
+
+export const ComponentsWrapper = ({ condition, ReturnComponent, AlternativeComponent }: IComponentsWrapper) => {
```

Comment:
```suggestion
export const LoadingWrappper = ({ isLoading, component, LoadingComponent }: ILoadingWrapper ) => {
```

I guess we don't need to have super reusable HOC, this HOC will be here only for loading purposes. so lets rename based on loading staff.